### ISSUE TITLE
connection of NDI polaris vicra camera via PLUS 

### ISSUE THREAD

**Issue by drclauzi** [PlusToolkit/PlusLib#761]
Hi there, I'm working with Slicer in combination with the NDI polaris vicra motion trackers to do navigation. We've been working with the PLUS toolkit in the last 6 months to connect the camera to Slicer but recently it doesn't connect any more. Is there anything that changed? I work with a Dell PC and Windows 10. Please let me know if you need any more information, thanks.

**Comment by adamrankin** [comment]
No changes. What does the error log say?

**Comment by lassoan** [comment]
Which Plus and a Slicer versions do you use?

**Comment by drclauzi** [comment]
Dear Andras, dear Adam,

Thanks for your swift response.
The Slicer version is 4.11. and I think the Plus version is 2.8 or 2.3.
 I cannot tell you anything about the error message at the moment, as I
don't remember if exactly and  I don't have the camera here with me.
I can connect again tomorrow when I'm in the lab and will then let you know
about the error message.

Thanks again for your help,
Claudia

**Comment by drclauzi** [comment]
Hi there again,

So, the error message is "connection failed, please select another device set and try again"
In th PLUS code it says: "error 018.108000,/ failed to start server process" I also had error messages like:
"018.107000- data collector failed to connect to devices"
"018.107000-unable to connect device: tracker device" 
"018.107000-tracker device: cannot connecr to data source, ConnectInternal failed"
"018.106000-failed to open port"

One of the PhD students very experienced with slicer has written the navigation code in slicer for the project and we've been working together closely. He checked everything related to slicer and said it is not a slicer problem, but rather a connection problem between Plus and the camera - we're using IGTlink. It might be the port, but I only have one USB port in this computer and the trackers are connecting fine with the NDI software if I use the camera with teh trackers and the computer on its own. The laptop only has USB-C ports and I connect the camera via the USB port in the adapter. 
We deinstalled Plus already and reinstalled it, but the same error messages are coming up...

Any ideas?
Thank you so much!!

**Comment by ungi** [comment]
Could you please copy the contents of your PLUS config file here? I haven't used this kind of tracker in a while, but I think there was a port number in the config file that we needed to check against the device port number in the Device Manager of Windows. Or something like that. Sometimes that port number can change and the config file needs to be updated with the new number.

**Comment by adamrankin** [comment]
I recommend removing the port number from the config file altogether. Plus will autofind any connected NDI cameras.

**Comment by drclauzi** [comment]
<PlusConfiguration version="2.1">
  <DataCollection StartupDelaySec="1.0" >
    <DeviceSet 
      Name="CANINEHIPNAV PlusServer: NDI Polaris tracker with passive markers" 
      Description="Broadcasting tool tracking data through OpenIGTLink For NDI Polaris passive marker starting kit: Reamer (8700338), Stylus (8700340), Impactor (8700339), and Reference (8700449)" />
    <Device
      Id="TrackerDevice"
      Type="PolarisTracker"
      ToolReferenceFrame="Tracker" >
      <DataSources>
        <DataSource Type="Tool" Id="Reamer" RomFile="NdiToolDefinitions/8700338.rom"  />
        <DataSource Type="Tool" Id="Stylus" RomFile="NdiToolDefinitions/8700340.rom"  />
        <DataSource Type="Tool" Id="Impactor" RomFile="NdiToolDefinitions/8700339.rom"  />
        <DataSource Type="Tool" Id="Reference" RomFile="NdiToolDefinitions/8700449.rom"  />
      </DataSources>
      <OutputChannels>
        <OutputChannel Id="TrackerStream" >
          <DataSource Id="Reamer"/>
          <DataSource Id="Stylus"/>
          <DataSource Id="Impactor"/>
          <DataSource Id="Reference"/>
        </OutputChannel>
      </OutputChannels>
    </Device>
  </DataCollection>
  <PlusOpenIGTLinkServer 
    MaxNumberOfIgtlMessagesToSend="1"
    MaxTimeSpentWithProcessingMs="50"
    ListeningPort="18944"
    SendValidTransformsOnly="true"
    OutputChannelId="TrackerStream" >
    <DefaultClientInfo>
      <MessageTypes>
        <Message Type="TRANSFORM" />
      </MessageTypes>
      <TransformNames>
        <Transform Name="StylusToReference" />
        <Transform Name="ReamerToReference" />
        <Transform Name="ImpactorToReference" />
      </TransformNames>
    </DefaultClientInfo>
  </PlusOpenIGTLinkServer>
</PlusConfiguration>

**Comment by zacbaum** [comment]
Hi @ungi, @adamrankin, and @lassoan - here's the config file we've been using. As @drclauzi mentioned - it worked previously with identical hardware and PLUS version (2.3.x).

```
<PlusConfiguration version="2.1">
  <DataCollection StartupDelaySec="1.0" >
    <DeviceSet 
      Name="CANINEHIPNAV PlusServer: NDI Polaris tracker with passive markers" 
      Description="Broadcasting tool tracking data through OpenIGTLink For NDI Polaris passive marker starting kit: Reamer (8700338), Stylus (8700340), Impactor (8700339), and Reference (8700449)" />
    <Device
      Id="TrackerDevice"
      Type="PolarisTracker"
      ToolReferenceFrame="Tracker" >
      <DataSources>
        <DataSource Type="Tool" Id="Reamer" RomFile="NdiToolDefinitions/8700338.rom"  />
        <DataSource Type="Tool" Id="Stylus" RomFile="NdiToolDefinitions/8700340.rom"  />
        <DataSource Type="Tool" Id="Impactor" RomFile="NdiToolDefinitions/8700339.rom"  />
        <DataSource Type="Tool" Id="Reference" RomFile="NdiToolDefinitions/8700449.rom"  />
      </DataSources>
      <OutputChannels>
        <OutputChannel Id="TrackerStream" >
          <DataSource Id="Reamer"/>
          <DataSource Id="Stylus"/>
          <DataSource Id="Impactor"/>
          <DataSource Id="Reference"/>
        </OutputChannel>
      </OutputChannels>
    </Device>
  </DataCollection>
  <PlusOpenIGTLinkServer 
    MaxNumberOfIgtlMessagesToSend="1"
    MaxTimeSpentWithProcessingMs="50"
    ListeningPort="18944"
    SendValidTransformsOnly="true"
    OutputChannelId="TrackerStream" >
    <DefaultClientInfo>
      <MessageTypes>
        <Message Type="TRANSFORM" />
      </MessageTypes>
      <TransformNames>
        <Transform Name="StylusToReference" />
        <Transform Name="ReamerToReference" />
        <Transform Name="ImpactorToReference" />
      </TransformNames>
    </DefaultClientInfo>
  </PlusOpenIGTLinkServer>
</PlusConfiguration>
```

**Comment by adamrankin** [comment]
Does the tracker work with Track software? Is the Track software running when you try to connect from Plus?

**Comment by drclauzi** [comment]
The trackers work fine and are picked up by the camera when Iused it with the NDI software.
No, the trackers are not picked up - no connection can be established (according to the plus error message) with PLUS. I don't know if I understand the question correctly ...

**Comment by adamrankin** [comment]
If the NDI software is connected to the camera, Plus cannot connect. The safest option is to ensure that no NDI software is running when you try to connect from Plus.

**Comment by drclauzi** [comment]
I found that out ;-) 
The NDI software is not connected when I work with Plus and slicer

**Comment by drclauzi** [comment]
Hello everybody, any other ideas what it could be? Is it possible that it is not compatible with the NDI software any more? Thanks for your help

**Comment by drclauzi** [comment]
Yes, I understand that the software should not run at the same time, but is it possible that there is an "issue" regarding NDI and the camera per se?

**Comment by ungi** [comment]
I met with @drclauzi to investigate. We had limited time, so I'm not entirely sure what solved the problem. But we did these three things, which I just list here for the record:
- Look up the COM port in Device Manager and specify it directly in the config file. This made the tracker beep at least, but still no data streaming.
- We updated to the latest stable PLUS.
- We uninstalled Avast antivirus, which I think was somehow blocking PLUS.
Then, tracking went back to normal.
